reactive/archive_auth_mirror.py

```diff
@@ -1,14 +1,18 @@
 from charmhelpers.core import hookenv
 
-from charms.reactive import when, when_not
+from charms.reactive import when, when_not, hook
 
 from charms.layer.nginx import configure_site
 from charms.archive_auth_mirror import gpg, reprepro, utils
 
 
+@hook('install')
+def install():
+    utils.install_resources()
```

Comment:
You should also set a state and have the other handlers depend on it, so that you really ensure that this gets run before the other handlers. Without that, you still have a race condition.